scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/config/SecurityContext.java

```diff
@@ -37,7 +37,8 @@ protected void configure(HttpSecurity http) throws Exception {
                 .antMatchers("/", "/resources/**",
                         "/register").permitAll()
                 // The rest of the our application is protected.
-                .antMatchers( "/users/**", "/admin/**").hasRole("ADMIN")
+                .antMatchers("/users/**", "/admin/**").hasRole("ADMIN")
+                .antMatchers("/datasets/**").hasRole("USER")
```

Comment:
I think I made this comment in another story before (maybe it's in one of the other branches?), but I would choose one prefix that is accessible by users with role `USER` and then use that for all parts of the application that require the user role (e.g. `/auth` or `/secured` or something). Your urls would then look like `/auth/datasets/...`, `/auth/projects/...`, etc. It's less error prone and doesn't require changes to the SecurityContext every time a new functionality is added.